### PR TITLE
rpc: support EXISTS operator in /tx_search query

### LINKED ISSUE #4763: Add EXISTS condition for tx_search
## Summary

Add EXISTS condition for `tx_search`.

## Problem Definition

The subscription feature supports EXISTS condition as the result of #4077. But `tx_search` is left unsupported.

To reproduce, just try on RPC:
```
http://localhost:26657/tx_search?query="dummy EXISTS"
```
It panics with: "other operators should be handled already".

## Proposal

Add support for EXISTS when searching for transactions similar to that of subscription.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by melekes**
Closes #4763

### CODE REVIEW

File: state/txindex/kv/kv.go [hunk 1]
```diff
@@ -426,6 +426,26 @@ func (txi *TxIndex) match(
...
+	case c.Op == query.OpExists:
+		// XXX: can't use startKeyBz here because c.Operand is nil
+		// (e.g. "account.owner/<nil>/" won't match w/ a single row)
+		it, err := dbm.IteratePrefix(txi.store, startKey(c.CompositeKey))
+		if err != nil {
+			panic(err)
+		}
+		defer it.Close()
+
+		for ; it.Valid(); it.Next() {
```
Comment (erikgrinaker) [inline state/txindex/kv/kv.go:438]: This should check `it.Error()` after the loop ends, e.g.:

```go
if err := itr.Error(); err != nil {
  panic(err)
}
```

Comment (melekes) [reply state/txindex/kv/kv.go:438]: what for?

Comment (erikgrinaker) [reply state/txindex/kv/kv.go:438]: So that we don't silently ignore errors, which would lead to clients getting incomplete results?

**Comment by melekes** [comment]
> do we have somewhere in the docs explaining the indexer? if so we should add this there

I can't seem find anything. I swear we had a description of query syntax somewhere. maybe it got removed during godoc -> swagger transition.

**Pull request merged**